app/src/main/java/com/kleshchin/danil/filemanager/MainActivity.java

```diff
@@ -0,0 +1,118 @@
+package com.kleshchin.danil.filemanager;
+
+import android.content.Intent;
+import android.graphics.PorterDuff;
+import android.graphics.drawable.Drawable;
+import android.net.Uri;
+import android.os.Bundle;
+import android.os.Parcelable;
+import android.support.v7.app.AppCompatActivity;
+import android.support.v7.widget.Toolbar;
+import android.view.MenuItem;
+import android.view.View;
+import android.webkit.MimeTypeMap;
+import android.widget.AdapterView;
+import android.widget.EditText;
+import android.widget.ListView;
+
+import java.io.File;
+
+public class MainActivity extends AppCompatActivity {
+    private ListView listView;
+    private ListAdapter listAdapter;
+    private String mainPath = "/storage";
+    private String currentDirectory;
```

Comment:
Wouldn't it be handier to have a current directory as a File class instance? You always can get a path from it and you don't need to create a new File instance every time you need it.